### PR TITLE
Update bachelors_degree_by_gender example.

### PR DESCRIPTION
**Pull request by anntzer**
Use plt.grid() instead of manually adding grid lines (this requires
slightly changing the xlims to have a similar spacing).

Use FuncFormatters instead of fixed tick labels, both for generality and so
that the interactive cursor position text is properly formatted.

Add a `show()` at the end.

### CODE REVIEW

File: examples/showcase/bachelors_degrees_by_gender.py [hunk 1]
```diff
@@ -30,22 +30,20 @@
...
-ax.get_xaxis().get_major_formatter().set_useOffset(False)
 
 # Make sure your axis ticks are large enough to be easily read.
 # You don't want your viewers squinting to read your plot.
 plt.xticks(range(1970, 2011, 10), fontsize=14)
-plt.yticks(range(0, 91, 10), ['{0}%'.format(x)
-                              for x in range(0, 91, 10)], fontsize=14)
+plt.yticks(range(0, 91, 10), fontsize=14)
+ax.xaxis.set_major_formatter(plt.FuncFormatter('{:.0f}'.format))
+ax.yaxis.set_major_formatter(plt.FuncFormatter('{:.0f}%'.format))
```
Comment (Kojoley) [inline examples/showcase/bachelors_degrees_by_gender.py]: You can simply use `'{:.0%}'`

Comment (anntzer) [inline examples/showcase/bachelors_degrees_by_gender.py]: No, because this would multiply the entries by 100 (they are in [0, 100], not [0, 1]).

Comment (Kojoley) [inline examples/showcase/bachelors_degrees_by_gender.py]: Ahh, yes, sorry.

**Comment by tacaswell** [comment]
backported to v2.x as be37777

**Pull request merged**